wxGridCellFloatEditor compile warning on ARM architectures

<!--

Please describe exactly how you build wxWidgets, including the full `configure`
command line and/or `make` command line if relevant.

Please attach the full build log, but feel free to quote the relevant parts of
it here.

When using `configure`, please also attach `config.log` file.

-->

### Description
<!-- Describe the problem here -->
I'm seeing a compile warning when building wx 3.2.1 on Debian Unstable, but only on ARM architectures (arm64, armhf, armel).

```
./src/generic/grideditors.cpp: In member function ‘virtual void wxGridCellFloatEditor::StartingKey(wxKeyEvent&)’:
./src/generic/grideditors.cpp:1052:21: warning: comparison of integer expressions of different signedness: ‘int’ and ‘wxChar’ {aka ‘wchar_t’} [-Wsign-compare]
 1052 |          || keycode == wxNumberFormatter::GetDecimalSeparator() )
      |             ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./src/generic/grideditors.cpp: In member function ‘virtual bool wxGridCellFloatEditor::IsAcceptedKey(wxKeyEvent&)’:
./src/generic/grideditors.cpp:1189:33: warning: comparison of integer expressions of different signedness: ‘const int’ and ‘wxChar’ {aka ‘wchar_t’} [-Wsign-compare]
 1189 |                         keycode == wxNumberFormatter::GetDecimalSeparator() ||
      |                         ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
The full build log is available here: https://buildd.debian.org/status/fetch.php?pkg=wxwidgets3.2&arch=arm64&ver=3.2.1%2Bdfsg-3&stamp=1672160163&raw=0


### Platform and version information

- wxWidgets version <!-- [e.g. 3.2.1] --> you are building: 3.2.1
- wxWidgets port <!-- [e.g. wxMSW, wxGTK, wxOSX] --> you are building: wxGTK
- OS <!-- [e.g. Windows 10, Ubuntu 22.10, macOS 15] --> and its version: Debian Unstable
- Compiler <!-- [e.g. MSVS 2022, gcc 12.1] --> being used: gcc 12.2.0

--- a/src/generic/grideditors.cpp
+++ b/src/generic/grideditors.cpp
@@ -855,7 +855,7 @@ bool wxGridCellNumberEditor::IsAcceptedKey(wxKeyEvent& event)
 
 void wxGridCellNumberEditor::StartingKey(wxKeyEvent& event)
 {
-    int keycode = event.GetKeyCode();
+    const wxChar keycode = event.GetUnicodeKey();
     if ( !HasRange() )
     {
         if ( wxIsdigit(keycode) || keycode == '+' || keycode == '-')
@@ -1022,7 +1022,7 @@ void wxGridCellFloatEditor::Reset()
 
 void wxGridCellFloatEditor::StartingKey(wxKeyEvent& event)
 {
-    int keycode = event.GetKeyCode();
+    const wxChar keycode = event.GetUnicodeKey();
 
     if ( wxIsdigit(keycode) || keycode == '+' || keycode == '-'
          || keycode == wxNumberFormatter::GetDecimalSeparator() )
@@ -1156,7 +1156,7 @@ bool wxGridCellFloatEditor::IsAcceptedKey(wxKeyEvent& event)
 {
     if ( wxGridCellEditor::IsAcceptedKey(event) )
     {
-        const int keycode = event.GetKeyCode();
+        const wxChar keycode = event.GetUnicodeKey();
         if ( wxIsascii(keycode) )
         {
             // accept digits, 'e' as in '1e+6', also '-', '+', and '.'
@@ -1351,8 +1351,7 @@ bool wxGridCellBoolEditor::IsAcceptedKey(wxKeyEvent& event)
 {
     if ( wxGridCellEditor::IsAcceptedKey(event) )
     {
-        int keycode = event.GetKeyCode();
-        switch ( keycode )
+        switch ( event.GetUnicodeKey() )
         {
             case WXK_SPACE:
             case '+':
@@ -1366,8 +1365,7 @@ bool wxGridCellBoolEditor::IsAcceptedKey(wxKeyEvent& event)
 
 void wxGridCellBoolEditor::StartingKey(wxKeyEvent& event)
 {
-    int keycode = event.GetKeyCode();
-    switch ( keycode )
+    switch ( event.GetUnicodeKey() )
     {
         case WXK_SPACE:
             CBox()->SetValue(!CBox()->GetValue());

